README format update

conviva-nsolanki opened the pull request:
Changes made:

- match the format with other platform. [Android](https://github.com/conviva-nsolanki/conviva-android-appanalytics/blob/main/README.md)
- move local storage information under client id synchronization section 
- move [Autocollection of Clicks](https://github.com/Conviva/conviva-js-appanalytics/blob/main/README.md#autocollection-of-clicks) under limitation. 
- move [AutoCollection of Network Request made using XMLHttpRequest and fetch api](https://github.com/Conviva/conviva-js-appanalytics/blob/main/README.md#autocollection-of-network-request-made-using-xmlhttprequest-and-fetch-api) under limitation
- change `"Example Values"` under the set device metadata
- add comment on where to initialize the sdk


Plan to move to FAQ:
- [Conviva Video Events to App Insight](https://github.com/Conviva/conviva-js-appanalytics/blob/main/README.md#conviva-video-events-to-app-insight)
- [Auto Ingestion of "traceparent" header to network requests](https://github.com/Conviva/conviva-js-appanalytics/blob/main/README.md#auto-ingestion-of-traceparent-header-to-network-requests)

conviva-nsolanki committed:
Update README.md

conviva-nsolanki committed:
Update README.md

kedarmarsada commented on README.md:
```diff
@@ -1,119 +1,103 @@
-  appVersion: "1.1.0"
-});
-clientId = getClientId();
+import { convivaAppTracker } from '@convivainc/conviva-js-appanalytics';
```
Can we have single import in "## Import to you project" section with mandatory APIs only Example: convivaAppTracker, trackPageView and setUserId
As a developer I will copy imports once & paste it. Later, I do not want to keep going back and importing additional unless there is a need for optional API to be imported?

conviva-nsolanki replied:
add the list of imports related to mandatory implementation. for optional feature import code snippet contains the import related to the optional feature. 
 
<img width="998" alt="Screenshot 2025-03-27 at 12 42 23 PM" src="https://github.com/user-attachments/assets/6edf1200-195e-40c6-99d6-7a55937b9c78" />

kedarmarsada commented on README.md:
```diff
@@ -1,119 +1,103 @@
-Auto collection for Errors / exceptions is enabled by default.
 
-Alternatively, you could report exceptions manually using the following API:
+### 5. Error Reporting 
```
Since this is auto collected. Let us remove from this section. 
We auto collect uncaught exceptions, unhandled rejections. 

Later we can have another section only for optional reporting of "caught exceptions" using trackError

conviva-nsolanki replied:
sounds good. I will move this to `More Features` section.

conviva-nsolanki replied:
<img width="1163" alt="Screenshot 2025-03-27 at 12 32 36 PM" src="https://github.com/user-attachments/assets/fdf848a0-053c-427e-9a72-9cf007106bf9" />

kedarmarsada commented on README.md:
```diff
@@ -284,60 +324,83 @@ deviceMetadata: An object containing the key-value pairs of pre-defined values f
+Event | Occurrence |
+------|------------|
+network_request | After receiving the network request response. [Refer limitations](#limitations). |
+page_ping | Max X and Y scroll positions difference comparing to the last event.|
```
Do we need to mention page_ping? As this is conviva internal right?

conviva-nsolanki replied:
good point. this was copied from the current doc and did not realize it. I will remove this. thank you!

kedarmarsada commented on README.md:
```diff
@@ -284,60 +324,83 @@ deviceMetadata: An object containing the key-value pairs of pre-defined values f
+### Limitations
+
+<details>
+  <summary><b>button_click, link_click</b></summary>
```
Heading can be just clicks.

conviva-nsolanki replied:
<img width="1037" alt="Screenshot 2025-03-27 at 12 33 30 PM" src="https://github.com/user-attachments/assets/9ac7c521-770e-4ce9-a25d-429b1782fa53" />

kedarmarsada reviewed (changes requested):
Overall, I got a feel that we have split the sections as "Mandatory first" (To get all out of box metrics in minimal steps), "Optional next".
Assuming this, I have given my review coments. please check.

conviva-nsolanki committed:
Update README.md

- add list of imports for required package
- move error reporting to more feature section 
- remove page_ping from auto-collected events table 
- change button_click, link_click limitation to simply Click 
- update sample snippet for setCustomTag and unsetCustomTag

conviva-nsolanki commented:
Hi @kedarmarsada, 
from this [reporting](https://github.com/Conviva/conviva-js-appanalytics/issues/20), the code snippet for custom tags is not correct. also, I believe `unsetCustomTag()` function need some fixes. I will file a support ticket for `unsetCustomTag()`. 

here is updated version of custom tags related code snippets & please confirm: 
<img width="986" alt="Screenshot 2025-03-27 at 12 36 43 PM" src="https://github.com/user-attachments/assets/e71229bf-3f1f-436b-abe4-5ad8e32cbe88" />
